js/client-mainmenu.js

```diff
@@ -1122,12 +1122,60 @@
 					for (var i = 0; i < teams.length; i++) {
 						if ((!teams[i].format && !teamFormat) || teams[i].format === teamFormat) {
 							var selected = (i === curTeam);
-							bufs[curBuf] += '<li><button name="selectTeam" value="' + i + '"' + (selected ? ' class="sel"' : '') + '>' + BattleLog.escapeHTML(teams[i].name) + '</button></li>';
-							count++;
-							if (count % bufBoundary == 0 && curBuf < 4) curBuf++;
+							if (this.folderToggleOn !== true) {
+								bufs[curBuf] += '<li><button name="selectTeam" value="' + i + '"' + (selected ? ' class="sel"' : '') + '>' + BattleLog.escapeHTML(teams[i].name) + '</button></li>';
+								count++;
+								if (count % bufBoundary == 0 && curBuf < 4) curBuf++;
+							} else {
+								var folderName = teams[i].folder || "";
+								if (folderName !== "") {
+									if (folders[folderName] === undefined) folders[folderName] = []; // create empty array to push to
+									var thisTeam = teams[i];
+									thisTeam.id = i; // create this ID for future sorting purposes, with respect to storage
+									folders[folderName].push(thisTeam); // shows the folders with only the content of format, move out if needed
+								}
+							}
 						}
 					}
+					if (this.folderToggleOn) {
+						var keys = Object.keys(folders);
+						for (var i = 0; i < keys.length; i++) {
```

Comment:
Do we ban `for..in`?

Reply:
Oh, nice javascript. I tried forEach and it failed so I gave up on all the iterable stuff. Going to implement that now... thanks

Reply:
...

Reply:
I'm still confused as to why this isn't `for (var key in folders)` ?